table/tables/tables.go

```diff
@@ -1208,7 +1208,7 @@ func (t *TableCommon) removeRowIndices(ctx sessionctx.Context, h kv.Handle, rec
 		return err
 	}
 	for _, v := range t.deletableIndices() {
-		if v.Meta().Primary && (t.Meta().IsCommonHandle || t.Meta().PKIsHandle) {
```

Comment:
Why remove `t.Meta().PKIsHandle` ?

Reply:
When `PKIsHandle` is true, `t.indiecs` doesn't contain the primary index, according to @lysu

Reply:
Removing this condition directly adds an implicit dependency that could be broken by other changes, it's a bit dangenrous to do this. 
To solve this complexity, we could abstract this related judege code into a single function and add related comments for it, also we will need to verify if there's related check logic or test cases to cover the constraint.

Reply:
OK makes sense. The current condition can mislead readers about the implicit contract, but also makes it safer against changes.
I'm gonna leave it as is for now.